src/io/flutter/actions/FlutterSdkAction.java

```diff
@@ -30,19 +32,36 @@
   public void actionPerformed(AnActionEvent event) {
     final Project project = DumbAwareAction.getEventProject(event);
 
-    final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
-    if (sdk == null) {
-      showMissingSdkDialog(project);
-      return;
-    }
+    if(enableActionInBazelContext() && project != null) {
+      // See if the Bazel workspace provides a script.
+      final Workspace workspace = WorkspaceCache.getInstance(project).getNow();
+      if (workspace != null) {
+          FlutterInitializer.sendAnalyticsAction(this);
+          FileDocumentManager.getInstance().saveAllDocuments();
+          startCommandInBazelContext(project, workspace);
+      }
+    } else {
+      final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
+      if (sdk == null) {
+        showMissingSdkDialog(project);
+        return;
+      }
 
-    FlutterInitializer.sendAnalyticsAction(this);
-    FileDocumentManager.getInstance().saveAllDocuments();
-    startCommand(project, sdk, PubRoot.forEventWithRefresh(event));
+      FlutterInitializer.sendAnalyticsAction(this);
+      FileDocumentManager.getInstance().saveAllDocuments();
+      startCommand(project, sdk, PubRoot.forEventWithRefresh(event));
+    }
   }
 
   public abstract void startCommand(@NotNull Project project, @NotNull FlutterSdk sdk, @Nullable PubRoot root);
 
+  public void startCommandInBazelContext(@NotNull Project project, @NotNull Workspace workspace) {
+  }
```

Comment:
Maybe make this abstract?  Or if it's only appropriate for `FlutterDoctorAction`s, consider putting the conditional logic in there?